### PR TITLE
CLI/Import - test_bz1160847_translate_macros fixed

### PR DESCRIPTION
**Pull request by rplevka**
The modification of this test unit was necessary to mimic the new behavior of the `import config-file` command (see [BZ1226981](https://bugzilla.redhat.com/show_bug.cgi?id=1226981#c4))

```bash
$ nosetests -m test_bz1160847_translate_macros test_import.py -vs
@test: Check whether all supported Sat5 macros are being ... 

ok

----------------------------------------------------------------------
Ran 1 test in 27.507s

OK
```

### CODE REVIEW

**Comment by Ichimonji10** [comment]
ACK

File: tests/foreman/cli/test_import.py [hunk 1]
```diff
@@ -66,7 +67,9 @@ def build_csv_file(rows=None, dirname=None):
         )
         csv_writer.writeheader()
         for row in rows:
-            csv_writer.writerow(row)
+            csv_writer.writerow({
+                key: val.encode('utf8') for key, val in row.items()
```
Comment (elyezer) [inline tests/foreman/cli/test_import.py]: Using dict comprehension we are explicitly saying that we does not support python 2.6. I completely agree on this :smile:

File: tests/foreman/cli/test_import.py [hunk 2]
```diff
@@ -149,6 +152,14 @@ def gen_import_org_data():
             u'organization': gen_string('alphanumeric')
         }
         for i in range(3)
+    ],
```
Comment (elyezer) [inline tests/foreman/cli/test_import.py]: Join line 156 here: `], [` or unindent 156.

File: tests/foreman/cli/test_import.py [hunk 3]
```diff
@@ -163,6 +174,15 @@ def gen_import_user_data():
             u'username': gen_string('alphanumeric')
         }
         for i in range(3)
+    ],
+        [
```
Comment (elyezer) [inline tests/foreman/cli/test_import.py]: Same as previous comment.

File: tests/foreman/cli/test_import.py [hunk 4]
```diff
@@ -1226,7 +1246,7 @@ def test_bz1160847_translate_macros(self):
             u'channel_id': u'3',
             u'channel': u'config-1',
             u'channel_type': u'normal',
-            u'path': u'/etc/sysconfig/rhn/systemid',
+            u'path': gen_string('utf8') + gen_string('alphanumeric'),
```
Comment (elyezer) [inline tests/foreman/cli/test_import.py]: So no slash in the path anymore?

File: tests/foreman/cli/test_import.py [hunk 5]
```diff
@@ -1274,10 +1294,16 @@ def test_bz1160847_translate_macros(self):
                 u'generate-only': True,
             }).return_code, 0
         )
+        prefix = re.sub(invalid_chars, '', org['name'])
+        erb_file = re.sub(invalid_chars, '', csv_row['path'])
+        if len(prefix) == 0:
+            prefix = u'orgid' + org['id']
+        if len(erb_file) == 0:
+            prefix = u'orgid' + org['id']
```
Comment (elyezer) [inline tests/foreman/cli/test_import.py]: s/prefix/erb_file

**Comment by elyezer** [comment]
ACK, pending comments.

Feel free to rebase :smile:

File: tests/foreman/cli/test_import.py (hunk 4 above)
Comment (rplevka) [inline tests/foreman/cli/test_import.py]: the path does not matter in this case. It is being converted to a single file name,
sp e.g. ```u'/etc/sysconfig/rhn/systemid'``` will become `etc_system_sysconfig_rhn_systemid.erb`
there is a rule for removing 'ugly' characters (```[^\da-zA-Z\-\.\_]```)
so the test is just to take some string containing `nice` and `ugly` characters.

**Comment by rplevka** [comment]
comments addressed.

File: tests/foreman/cli/test_import.py (hunk 4 above)
Comment (elyezer) [inline tests/foreman/cli/test_import.py]: thanks

**Comment by elyezer** [comment]
ACK

**Pull request merged**